### ISSUE TITLE
End2End AMP training

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#1529]
https://github.com/pytorch/torchdynamo/blob/main/benchmarks/training_loss.py performs end-to-end training example. However, we are not testing AMP training in a standard manner. 

AMP training has GradScaler which performs scaling, monitors gradients, discards batch sizes if gradients are inf/nan. More importantly, GradScaler is stateful. All this warrants an end to end training testing with TorchDynamo and TorchInductor stack.

There would be many great articles on AMP. This is the first one I stumbled upon - https://spell.ml/blog/mixed-precision-training-with-pytorch-Xuk7YBEAACAASJam

cc @yanboliang 
cc @xuzhao9 if torchbench already has AMP setup for end to end training

cc @msaroufim if you are interested in this
